src/array_value.cpp

```diff
@@ -7,6 +7,14 @@
 
 namespace Natalie {
 
+ValuePtr ArrayValue::allocate(Env *env, size_t argc, ValuePtr *args) {
+    if (argc > 0)
+        env->raise("ArgumentError",
+            "wrong number of arguments (given {}, expected 0)",
+            argc);
```

Comment:
We have some convenience methods for this pattern: https://github.com/seven1m/natalie/blob/03602356b9fd89a30e79c00dba855b2b40477ab5/include/natalie/env.hpp#L67-L70

Reply:
oh that's pretty neat!